server/routes.go

```diff
@@ -136,10 +138,34 @@ func (r *routesImpl) SetDefaultRoute(backend string) {
 	}).Info("Using default route")
 }
 
+func (r *routesImpl) SimplifySRV(srvEnabled bool) {
+	r.simplifySRV = srvEnabled
+}
+
 func (r *routesImpl) FindBackendForServerAddress(ctx context.Context, serverAddress string) (string, string, func(ctx context.Context) error) {
 	r.RLock()
 	defer r.RUnlock()
 
+	if r.simplifySRV {
+		serverAddress = strings.TrimSuffix(serverAddress, ".")
+		parts := strings.Split(serverAddress, ".")
+		tcpIndex := -1
+		for i, part := range parts {
+			if part == "_tcp" {
+				tcpIndex = i
+				break
+			}
+		}
+		if tcpIndex != -1 {
+			parts = parts[tcpIndex+1:]
+		}
+
+		if len(parts) > 0 && parts[len(parts)-1] == "_tcp" {
+			parts = parts[:len(parts)-1]
```

Comment:
then nothing its logic wont happen, 

because the logic goes as "look in index for _tcp and remove it and everything before it"

if _tcp is not found, then tcpIndex remains -1, and then the parts are put back together, 
if there is no index, there is no reason to remove, 

atleast thats how it should work,

Reply:
let me test it actually, see if its actually needed